app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,58 @@
+class PostsController < ApplicationController
+  def index
```

Comment:
we have to use a dry concept here

add a filter method
before_action :set_post, only: %i[ show edit update destroy ]

add private method
def set_user
@post = Post.find(params[:id])
end
remove the repeating code from show edit update destroy methods

Reply:
applied DRY principle.